Pull Request para entrega 2

ffigueroacasas opened the pull request:
Cambios introducidos:
Visibles:
- le dimos un estilo similar a la mayoria de los botones
- eliminamos botones superpuestos y flotantes
- eliminamos todos los tipos any posibles
No visibles:
- eliminamos abreviaciones, variables en español y console.logs innecesarios
- Subdividimos el app component en subcomponentes mas pequeños
- eliminamos los estilos en linea e introdujimos variables scss

P-garrido committed:
agregado de navbar

Agustin-Caucino committed:
hice un navbar y componentes inicio y productos

Agustin-Caucino committed:
Agregue Login Form

Agustin-Caucino committed:
Actualice componente login

Agustin-Caucino committed:
Merge pull request #1 from P-garrido/agus

Agus

P-garrido committed:
Agregado de clases

Agustin-Caucino committed:
Agregue Cart component

Agustin-Caucino committed:
Avance con productos y cart components

P-garrido committed:
agregado de footer e inicio

Agustin-Caucino committed:
agregue products service

Agustin-Caucino committed:
Merge pull request #2 from P-garrido/LogIn

agregue products service

Agustin-Caucino committed:
actualice login

Agustin-Caucino committed:
Merge pull request #3 from P-garrido/agus

Agus

Agustin-Caucino committed:
agregue opcion de agregar productos

P-garrido committed:
Página básica de servicios (falta terminar)

Agustin-Caucino committed:
Conecte componentes productos, productos servicios y new-product con la base de datos

P-garrido committed:
Cambio estetico servicios

P-garrido committed:
segui con servicios

P-garrido committed:
Agregado de funcion addService

P-garrido committed:
Arranque con eliminar servicio

ffigueroacasas committed:
Agregados UsersComponent, SignUpComponent
y UsersService

P-garrido committed:
termine eliminar servicios pero no me sale editar

P-garrido committed:
Arreglo de errores

Agustin-Caucino committed:
mejore products component

Agustin-Caucino committed:
Agregue administracion de productos

ffigueroacasas committed:
Completado CRUD Usuarios. Agregadas validaciones a
SignUpComponent.

Agustin-Caucino committed:
agregue editar producto

P-garrido committed:
Termine editar servicio

P-garrido committed:
Cambios de estética y arranque contratar servicio

Agustin-Caucino committed:
agregue funcionalidad al carrito y empece a hacer la compra de productos

P-garrido committed:
Termine contratar servicio (falta validar usuario)

ffigueroacasas committed:
Modificaciones chicas en LogInComponent y class.ts
Modificaciones chicas en UsersComponent
Agregado de filtro por tipo de usuario en UsersComponent

Agustin-Caucino committed:
termine de registrar los pedidos

Agustin-Caucino committed:
Empece a componentizar productos y carrito

ffigueroacasas committed:
Componente Usuarios hecho responsive

P-garrido committed:
barra de busqueda y validacion admin

P-garrido committed:
Arranque con la ventana servicios contratados

P-garrido committed:
Arranque con insertar horas de servicio

P-garrido committed:
Terminé editar cant_horas en clientes_servicios

ffigueroacasas committed:
Arreglado un bug cuando editabas usuario desde el UserComponent.

P-garrido committed:
Hice responsive la parte de mantenimiento

P-garrido committed:
Hice "responsive" la parte de mantenimiento

P-garrido committed:
Retoques en formularios no validos

P-garrido committed:
Agregué la funcionalidad de enviar mensaje

P-garrido committed:
Agregué descripcion y acordeon

Agustin-Caucino committed:
componentice admin-products

P-garrido committed:
Arranque con el token

P-garrido committed:
Seguí con el token y arranque login

P-garrido committed:
Mejore login y valide algunas acciones

P-garrido committed:
Puse el app component como main

P-garrido committed:
Verificacion admin en navbar

P-garrido committed:
Merge branch 'main' into pedro

P-garrido committed:
Merge pull request #4 from P-garrido/pedro

Pedro

P-garrido committed:
Merge branch 'main' into ProductItem

P-garrido committed:
Merge pull request #5 from P-garrido/ProductItem

Product item

P-garrido committed:
Merge branch 'main' into Usuarios

P-garrido committed:
Merge pull request #6 from P-garrido/Usuarios

Usuarios

P-garrido committed:
Correccion de errores

Agustin-Caucino committed:
mejore listado pedidos

Agustin-Caucino committed:
Merge branch 'main' into ProductItem2

Agustin-Caucino committed:
Merge pull request #7 from P-garrido/ProductItem2

mejore listado pedidos

Agustin-Caucino committed:
Mejore footer

P-garrido committed:
Agregué ruta por defecto a inicio

Agustin-Caucino committed:
modifique unas cositas

Agustin-Caucino committed:
Merge pull request #8 from P-garrido/Agus2

modifique unas cositas

Agustin-Caucino committed:
modifique componente Log in y navbar

Agustin-Caucino committed:
Merge pull request #9 from P-garrido/Agus3

modifique componente Log in y navbar

P-garrido committed:
Maneje error en la compra de servicio

Agustin-Caucino committed:
cambie el subscribe del log in al componente log in

Agustin-Caucino committed:
Merge pull request #10 from P-garrido/Agus4

cambie el subscribe del log in al componente log in

Agustin-Caucino committed:
hice menu responsive (hamburguer menu)

Agustin-Caucino committed:
Merge pull request #11 from P-garrido/Agus5

hice menu responsive (hamburguer menu)

P-garrido committed:
agregue asociaciones user y serv

P-garrido committed:
Merge pull request #12 from P-garrido/pedro2

agregue asociaciones user y serv

P-garrido committed:
agregue estilo para servicio no realizado

P-garrido committed:
Merge pull request #13 from P-garrido/pedro-serv-no-realizado

agregue estilo para servicio no realizado

Agustin-Caucino committed:
modifique componente inicio y maneje error en el logIn

Agustin-Caucino committed:
Merge branch 'main' into inicioM

Agustin-Caucino committed:
Merge pull request #14 from P-garrido/inicioM

modifique componente inicio y maneje error en el logIn

P-garrido committed:
corregí errores en editar y borrar servicios

P-garrido committed:
Merge pull request #15 from P-garrido/pedro-serv

corregí errores en editar y borrar servicios

ffigueroacasas committed:
cambie para que el cliente sea 0 y el empleado 1.
cambie para que se limpie el form cuando vas a agregar un usuario despues de haber editado uno.

ffigueroacasas committed:
agregado front de sucursales

P-garrido committed:
Merge pull request #16 from P-garrido/larafront

agregado front de sucursales

P-garrido committed:
mobile inicio

P-garrido committed:
cambie navbar

P-garrido committed:
Agregué filtro en servicios-contratados

P-garrido committed:
Merge pull request #17 from P-garrido/filtro-servicios

Agregué filtro en servicios-contratados

P-garrido committed:
falta corregir error en el li:hover

P-garrido committed:
Arregle problema hover

P-garrido committed:
Merge pull request #18 from P-garrido/nav-bar-admin

Nav bar admin

P-garrido committed:
Ahora el logo redirige a inicio

P-garrido committed:
Arranque ocultar contra

P-garrido committed:
Termine ocultar contra

P-garrido committed:
Merge pull request #19 from P-garrido/login-oculta-contra

Login oculta contra

P-garrido committed:
Mejore modales en servicios

P-garrido committed:
Merge pull request #20 from P-garrido/mejora-servicios

Mejore modales en servicios

P-garrido committed:
Guardo datos usuario

P-garrido committed:
Merge pull request #21 from P-garrido/sessionStorage

Guardo datos usuario

P-garrido committed:
cambie localStorage por sessionStorage

ffigueroacasas committed:
Cambie un identificador en el componente usuarios
que no permitia eliminarlos

Agustin-Caucino committed:
agg subir imagenes al servidor y obtenerlas de ahi

Agustin-Caucino committed:
agregue q se vea el cliente en el pedido

ffigueroacasas committed:
Update README.md

ffigueroacasas committed:
detalles que me habian quedado en la compu

ffigueroacasas committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-front

P-garrido committed:
correcciones entrega

Agustin-Caucino committed:
corregi lo que dijo otaduy: cambiar any, no abreb
iar, componentizar cart, estilos en linea

Agustin-Caucino committed:
Merge pull request #25 from P-garrido/correccionesAgus

corregi lo que dijo otaduy: cambiar any, no abrev,

P-garrido committed:
Merge pull request #26 from P-garrido/pedroEntrega

correcciones entrega

P-garrido committed:
arranque cambios esteticos

P-garrido committed:
Estandarice cosas en servicios y login

P-garrido committed:
Merge pull request #27 from P-garrido/pedroCorrecciones

Pedro correcciones

P-garrido committed:
Modularice app module

ffigueroacasas committed:
Eliminadas abreviaciones y homogeneizados botones.
Agregada variable scss

ffigueroacasas committed:
Merge pull request #28 from P-garrido/CambiosFran

Cambios a componentes usuarios y signup

ffigueroacasas committed:
Acomodado el boton de check out

ffigueroacasas committed:
Merge pull request #29 from P-garrido/main

Actualizo mi rama de cambios

ffigueroacasas committed:
Acomode botones flotantes, borre algunos console.log()

ffigueroacasas committed:
Merge pull request #30 from P-garrido/CambiosFran

Ultimos cambios al componente usuarios

P-garrido committed:
a

P-garrido committed:
a

P-garrido committed:
Merge pull request #31 from P-garrido/modularizacionNgModules

Modularizacion ng modules

P-garrido committed:
Arregle problema de link registrarse

Agustin-Caucino committed:
pase filtro de productos al back

Agustin-Caucino committed:
Merge pull request #32 from P-garrido/correccionesAgus

pase filtro de productos al back

ffigueroacasas committed:
borre un if redundante

aotaduy commented on src/app/admin-product-item/admin-product-item.component.html:
```diff
@@ -0,0 +1,31 @@
+  >
+    <div class="user-flex-item">
+      <p class="user-flex-text">
+        Nombre de Producto: {{ product.nombre_producto }}
```
no es muy comun el uso de _ en vars de javascript lo mas comun es camelCase y creo que es mejor todo el codigo en ingles

aotaduy commented on src/app/admin-product-item/admin-product-item.component.ts:
```diff
@@ -0,0 +1,20 @@
+  styleUrls: ['./admin-product-item.component.scss']
+})
+export class AdminProductItemComponent {
+  @Input() product: any = {};
```
no es un Product? en vez de any

aotaduy commented on src/app/admin-products/admin-products.component.ts:
```diff
@@ -0,0 +1,32 @@
+    this.getAllProducts()
+  }
+  getAllProducts() {
+    this.prodService.loadProducts().subscribe((response: any) => this.productsList = response)
```
capaz estaria bnueno mostrar un loading mientras tanto

aotaduy commented on src/app/admin-services/admin-services.component.ts:
```diff
@@ -0,0 +1,53 @@
+
+
+  getAllServices() {
+    this.servicesList.splice(0, this.servicesList.length);
```
tal vez seria mejor reasignar servicesList con un arreglo nuevo

aotaduy commented on src/app/admin-services/admin-services.component.ts:
```diff
@@ -0,0 +1,53 @@
+    this.servicesList.splice(0, this.servicesList.length);
+    this.servicesService.getAllServices().subscribe(response => {
+      response.forEach((serv: any) => {
+        this.servicesList.push(new Service(serv.id_servicio, serv.desc_servicio, serv.precio_por_hora, serv.descripcion))
```
no podrias reemplazar el array entero seria mas eficiento que meter de a unoi los elementos

aotaduy commented on src/app/admin-services/admin-services.component.ts:
```diff
@@ -0,0 +1,53 @@
+  getAllServices() {
+    this.servicesList.splice(0, this.servicesList.length);
+    this.servicesService.getAllServices().subscribe(response => {
+      response.forEach((serv: any) => {
```
evitemos avbreviaciones

aotaduy commented on src/app/app.module.ts:
```diff
@@ -3,14 +3,96 @@ import { BrowserModule } from '@angular/platform-browser';
+
+
+// let auth: any[] = [LogInComponent, SignUpComponent, UsersComponent];
+// let admin: any[] = [NewProductComponent, AdminProductsComponent, ProductItemComponent, AdminProductItemComponent, NewServiceComponent, ServiciosContratadosComponent, ServContratadoComponent, SucursalesComponent, SucursalModalComponent, AdminServicesComponent, AdminServiceItemComponent];
```
eliminemos codigo comentado

aotaduy commented on src/app/branches.service.ts:
```diff
@@ -0,0 +1,73 @@
+
+  branches: Branch [] = []
+  api_schema: string = "http"
+  api_host: string = "localhost:1234"
```
esto lo deberiamos tomar de un environment asi lo podemos variar para deployarlo en algun server real

aotaduy commented on src/app/branches.service.ts:
```diff
@@ -0,0 +1,73 @@
+    this.http.delete(delete_url, {observe: 'response'}).subscribe(
+      response => {
+        if (response.status == 204) {
+          this.branches = this.branches.filter(o => o.id_sucursal != id);
```
evitemos abreviaciones, creo que el observe no hace falta ya que si es 204 o 200 funciona igal con un suscribe comun, no veo logica para otros status code que valga la pena discriminar

aotaduy commented on src/app/branches.service.ts:
```diff
@@ -0,0 +1,73 @@
+  update(branch: Branch): void {
+    if (branch.id_sucursal) {
+      let update_url = `${this.url_get_all_branches}/${branch.id_sucursal}/`;
+      this.http.patch<Branch>(update_url, branch, {observe: 'response'}).subscribe(
```
creo que esto se podria simplificar sin usar el observe response

aotaduy commented on src/app/branches.service.ts:
```diff
@@ -0,0 +1,73 @@
+      let update_url = `${this.url_get_all_branches}/${branch.id_sucursal}/`;
+      this.http.patch<Branch>(update_url, branch, {observe: 'response'}).subscribe(
+        (response) => {
+            let i = this.branches.findIndex(o => o.id_sucursal == branch.id_sucursal);
```
podria ser const

aotaduy commented on src/app/cart-product/cart-product.component.ts:
```diff
@@ -0,0 +1,25 @@
+export class CartProductComponent {
+
+  @Input() product!: Product;
+  @Output() remove = new EventEmitter();
```
deberiamos tipar los emiters

aotaduy commented on src/app/cart-product/cart-product.component.ts:
```diff
@@ -0,0 +1,25 @@
+  buttonAdd(pr:Product){
+    this.add.emit(pr)
+  } 
+  buttonRemove(pr:Product){
```
evitemos abreviaciones y el prefijo button creo que no hace falta ya que esta en todos los metodos

aotaduy commented on src/app/cart.service.ts:
```diff
@@ -0,0 +1,54 @@
+  }
+
+  addToOrder(orderId:number, idProd: number, cant: number){
+    return this.http.post('http://localhost:1234/order/products',
```
el host deberiamos obtenerlo de un env

aotaduy commented on src/app/footer/footer.component.html:
```diff
@@ -0,0 +1,150 @@
+          <!-- Grid column -->
+          <div class="col-md-2 col-lg-2 col-xl-2 mx-auto mt-3">
+            <h6 class="text-uppercase mb-4 font-weight-bold">Productos</h6>
+            <p>
```
veo datos hardcoded, eliminemos sinpo esta implementado

aotaduy commented on src/app/inicio/inicio.component.html:
```diff
@@ -0,0 +1,65 @@
+    <slide>
+      <img
+        src="assets/cloro-quimica.jpg"
+        alt="third slide"
```
codigo commentado mejor eliminar

aotaduy commented on src/app/log-in.service.ts:
```diff
@@ -0,0 +1,54 @@
+  readonly baseUrl = "http://localhost:1234/login";
+
+  token: any = "";
+  user: User = new User(-1, "", "", "", "", "", "", -1, "");
```
usemos siuempre las mismas comillas, simples o dobles pero de forma consistente

aotaduy commented on src/app/log-in.service.ts:
```diff
@@ -0,0 +1,54 @@
+
+  private handleError(error: HttpErrorResponse) {
+    if (error.status === 404) {
+      alert('Nombre de usuario o contrasena incorrectos');
```
no es lo ideal uisar el alert, puede bloquear el hilo del browser. 
deberiamos usr otro tipo de dialogo o alert ( toastr tal vez)

aotaduy commented on src/app/log-in/log-in.component.ts:
```diff
@@ -0,0 +1,52 @@
+  styleUrls: ['./log-in.component.scss']
+})
+export class LogInComponent {
+  logInForm = new FormGroup({
```
tal vez deberiamos usar un formGroup

aotaduy commented on src/app/models/classes.ts:
```diff
@@ -0,0 +1,131 @@
+  }
+}
+
+export class BoughtService {
```
deberiamos usar un archivo para cada clase

aotaduy commented on src/app/models/classes.ts:
```diff
@@ -0,0 +1,131 @@
+  email: string;
+
+
+  constructor(idUs: number, usNam: string, pass: string, fNam: string, lNam: string, ad: string, ph: string, type: number, em: string) {
```
en estos casos es mejor un param object https://refactoring.guru/es/introduce-parameter-object

aotaduy commented on src/app/nav-bar/nav-bar.component.scss:
```diff
@@ -0,0 +1,184 @@
+  .nav-links {
+    list-style: none;
+
+    li {
```
no se recomienda estilar tags

aotaduy commented on src/app/new-product/new-product.component.html:
```diff
@@ -0,0 +1,78 @@
+<div
+  style="
+    display: flex;
```
evitemos estilos inline, mejor usar una clase

aotaduy commented on src/app/new-product/new-product.component.ts:
```diff
@@ -0,0 +1,54 @@
+export class NewProductComponent {
+  proximo_id:Number | undefined
+  
+  newProductForm = new FormGroup({
```
mejor si podemos usar un formBuilder

aotaduy commented on src/app/new-product/new-product.component.ts:
```diff
@@ -0,0 +1,54 @@
+
+  constructor(private productService: ProductsService){
+    if(this.productService.productToEdit.nombre_producto != ''){
+      this.newProductForm.controls.productName.patchValue(this.productService.productToEdit.nombre_producto);
```
podrias hacer un patch de todo el group no campo a campo

aotaduy commented on src/app/order-item/order-item.component.ts:
```diff
@@ -0,0 +1,28 @@
+  constructor(private orderService: OrdersService){}
+
+  ngOnInit(): void {
+    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
```
eliminemos comentartis innecesarios

aotaduy commented on src/app/order-item/order-item.component.ts:
```diff
@@ -0,0 +1,28 @@
+    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
+    //Add 'implements OnInit' to the class.
+   for (let prod of this.order.productos_pedidos){
+    this.orderService.getProductById(prod.id_producto).subscribe((resp:any) => {
```
esto va a generar multiples llamadas, deberiamos tener un endpoint que puda manejar una coleccion

aotaduy commented on src/app/order-item/order-item.component.ts:
```diff
@@ -0,0 +1,28 @@
+   for (let prod of this.order.productos_pedidos){
+    this.orderService.getProductById(prod.id_producto).subscribe((resp:any) => {
+      this.products.push({
+        nombre_producto: resp.nombre_producto,
```
no parece muy bueno manejar por el nombre, deberiamos usar el id en todo caso

aotaduy commented on src/app/orders/orders.component.ts:
```diff
@@ -0,0 +1,36 @@
+  }
+  getOrderByProduct(){
+    this.ordersService.getProductByName(this.searchForm.value).subscribe((response:Product) => {
+      this.filteredOrders = this.orders.filter((o:any) => {
```
evitemos abreviaciones

aotaduy commented on src/app/product-item/product-item.component.ts:
```diff
@@ -0,0 +1,29 @@
+export class ProductItemComponent {
+
+
+  @Input() product: Product = {
```
no podrias hacer new Product() ?

aotaduy commented on src/app/serv-contratado/serv-contratado.component.ts:
```diff
@@ -0,0 +1,44 @@
+})
+export class ServContratadoComponent {
+
+  @Input() boughtService: BoughtService = new BoughtService(new Service(-1, "", -1, ""), new User(1, "", "", "", "", "", "", -1, ""), new Date(), -1, "");
```
tal vez podriamos ahcer un metrodo estatrico de BroughtService para crear estainstancia vacia

aotaduy commented on src/app/service/service.component.ts:
```diff
@@ -0,0 +1,68 @@
+  // }
+
+  // sendEdit() {
+  //   const newService = new EditServiceEvent(this.service.id!, this.editServiceForm);
```
codigo comentado deberiamos eliminar

aotaduy commented on src/app/services.service.ts:
```diff
@@ -0,0 +1,103 @@
+  deleteService(idService: number) {
+    const url = this.baseUrl + `/${idService}`;
+    const token = this.loginService.token;
+    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
```
veamos de agregar un interceptor para enviar el token en todos los requests

aotaduy commented on src/app/servicios-contratados/servicios-contratados.component.ts:
```diff
@@ -0,0 +1,79 @@
+    this.filteredBoughtServices.splice(0, this.filteredBoughtServices.length);
+    this.serviceService.getAllBoughtServices().subscribe(response => {
+      response.forEach((boughtService: any) => {
+        this.boughtServices.push(new BoughtService(new Service(boughtService.Servicio.id_servicio, boughtService.Servicio.desc_servicio, boughtService.Servicio.precio_por_hora, boughtService.Servicio.descripcion), new User(boughtService.Usuario.id_usuario, boughtService.Usuario.nombre_usuario, boughtService.Usuario.clave, boughtService.Usuario.nombre, boughtService.Usuario.apellido, boughtService ...
```
tratem,os de formater mejor este codigo

aotaduy commented on src/app/servicios-contratados/servicios-contratados.component.ts:
```diff
@@ -0,0 +1,79 @@
+    this.boughtServices.splice(0, this.boughtServices.length);
+    this.filteredBoughtServices.splice(0, this.filteredBoughtServices.length);
+    this.serviceService.getAllBoughtServices().subscribe(response => {
+      response.forEach((boughtService: any) => {
```
deberiamos usar un tipo apropiado

aotaduy reviewed (changes requested):
Una párte de los archivos no los llegue a ver pewro deberiamos corregir los problemas enumeraros en todos los archivos.

ffigueroacasas committed:
Movi las definiciones de clase a archivos propios

ffigueroacasas committed:
Cambie los imports para que funcione con las nuevas definiciones de clase

Agustin-Caucino committed:
correcciones: tipie los emitters, variables, add loading, puse base url en variable, borre coments etc

Agustin-Caucino committed:
Merge pull request #34 from P-garrido/correcionesAgus2

correcciones: tipie los emitters, variables, add loading, puse base u…

ffigueroacasas committed:
Unit test completo para Users component
Aunque la consola muestra datos de coverage distintos, el archivo de coverage muestra que todo el codigo esta cubierto por el test.

ffigueroacasas committed:
Merge pull request #35 from P-garrido/unitTesting

Unit test completo para Users component

Agustin-Caucino committed:
resolvi problemas de direccionamiento

ffigueroacasas committed:
resolvi problemas de direccionamiento (servicios)

ffigueroacasas committed:
Agregue jestCache a gitIgnore y lo saque de github

ffigueroacasas committed:
Agregue jestCache al gitignore y lo saque de git

aotaduy commented on src/app/admin-product-item/admin-product-item.component.html:
```diff
@@ -17,7 +17,7 @@
           class="btn btn-primary btn-editProd"
           routerLink="/addProduct"
-          (click)="editProduct(product)"
+          (click)="[editProduct(product), redirect()]"
```
creo qeu el redirect deberia ir adentro de editPRoduct, esta creacion de arreglo es algo artificial

aotaduy commented on src/app/admin-product-item/admin-product-item.component.ts:
```diff
@@ -10,11 +11,14 @@ export class AdminProductItemComponent {
     this.delete.emit(productToDelete);
   }
+  redirect(){
+    this.router.navigate(['/addProduct'])
```
no estoy seguro que esto corresponde aca tal vez deberia ir donde se escucha el evento

aotaduy commented on src/app/admin-services/admin-services.component.html:
```diff
@@ -1,5 +1,5 @@
 <div class="service-container">
-  <a class="btn btn-primary btn-add-serv" routerLink="/addService">
+  <a class="btn btn-primary btn-add-serv" routerLink="/addService" (click)="redirect()">
```
el routerLink ya deberia mandarlo a la otra pagina

aotaduy commented on src/app/models/boughtService.ts:
```diff
@@ -0,0 +1,18 @@
+  hourAmmount: number;
+  clientMessage: string;
+
+  constructor(serv: Service, us: User, servDate: Date, hourAm: number, cliMsj: string) {
```
evitemos abreviaciones como us

aotaduy commented on src/app/productos/productos.component.ts:
```diff
@@ -28,7 +29,8 @@ export class ProductosComponent implements OnInit {
   getAllProducts() {
-    this.productService.loadProducts().subscribe((response: Product[]) => {
+    this.loading = true;
+    this.productService.loadProducts().pipe( delay(1000),finalize(()=> this.loading=false)).subscribe((response: Product[]) => {
```
no esta muy bien la indent

aotaduy commented on src/app/users/users.component.ts:
```diff
@@ -86,14 +88,16 @@ export class UsersComponent implements  OnInit {
     this.edit = false
-    window.location.reload()
+    this.lastEditedUser = this.list.find(user => user.idUser === this.editId)
+    window.location.reload();
```
no es correcto llamar a windo de forma directa dificulta el unit test deberias injectarlo como localStorage

aotaduy commented on src/app/users/users.component.ts:
```diff
@@ -86,14 +88,16 @@ export class UsersComponent implements  OnInit {
     }
-    this.service.editUser(object, this.editId).subscribe()
+    this.service.editUser(object, this.editId)?.subscribe()
+    this.list.push(new User(this.editId, this.userForm.value.username as string, this.userForm.value.password as string, this.userForm.value.firstName as string, this.userForm.value.lastName as string, this.userForm.value.address as string, this.userForm.value.phoneNumber as string, this.userForm.value.userType as unknown as number, this.userForm.value.email as string));
```
porque lo tenes que agregar a la lista? no podrias recargar?

ffigueroacasas replied:
lo agrego a la lista para poder realizar el unit test sobre la funcion editUser, sino no estaba seguro como testearla/mockear la response de recargar la pagina. Como seria mejor testear el metodo sino? O seria preferible que no quede cubierta por el test?

aotaduy commented on src/app/users/users.component.ts:
```diff
@@ -86,14 +88,16 @@ export class UsersComponent implements  OnInit {
 
   addUser(){
-    const object: Object = {
+    const object: any = {
```
seguimos con any

aotaduy replied:
mejor usemos un nombre mas significartibo ademas de que Object con mayuscula es una clase del sistema

aotaduy commented on src/app/users/users.component.ts:
```diff
@@ -103,14 +107,16 @@ export class UsersComponent implements  OnInit {
-    this.service.getAllUsers().subscribe((users: User[])=> {
+    this.service.getAllUsers()?.subscribe((users: User[])=> {
       users.forEach((user: any) => {
         const object = new User(user.id_usuario, user.nombre_usuario, user.clave, user.nombre, user.apellido, user.direccion, user.telefono, user.tipo_usuario, user.email)
```
no creo que se anecesario pushear todos los usuarios no podes usar la lista que te llega y reem plazarla?, seria mas performante

ffigueroacasas replied:
como la lista me llega del backend (la base de datos esta en español, por lo que las propiedades quedan en español en el json), las propiedades de la clase User y de la response no coinciden, por eso construyo los users con el constructor y los pusheo para, en el resto del componente, tratarlos como Users con sus propiedades en ingles. Si asignara la lista directamente tendria que cambiar la definicion de User para que tenga las propiedades en español, que siento que no es preferible

ffigueroacasas committed:
hice algunos cambios sueltos
por ej. quitar tipo any y abreviaciones

Agustin-Caucino committed:
intente hacer lo de token

Agustin-Caucino committed:
agregue cargando pedidos, modifique getProductos en pedidos

Agustin-Caucino committed:
agregue handleError

Agustin-Caucino committed:
agregue handleError a orders

Agustin-Caucino committed:
termine lo del token

ffigueroacasas committed:
arregle bug de direccionamiento en editar servicios

ffigueroacasas committed:
borre codigo duplicado

ffigueroacasas committed:
resolvi bugs de actualizacion en usuarios y borre los ultimos console.logs

Agustin-Caucino committed:
puli algunas cosas de productos

ffigueroacasas committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-front

ffigueroacasas committed:
arreglado para que en el sign up solo registre clientes

P-garrido committed:
Saque función que no se usaba

P-garrido committed:
Merge branch 'main' of https://github.com/P-garrido/TP-dsw-front
